### ISSUE TITLE
Consider supporting zome only repositories

### ISSUE THREAD

**Issue by guillemcordoba** [holochain/scaffolding#249]
Currently the scaffolding tool only supports repositories with at least one `happ.yaml` and one `dna.yaml`. This covers the majority of projects, which are just web-happs that need to produce a `.webhapp` package to be published somewhere.

A lot of the repositories I work on don't fulfill that criteria, as they are just modules, so they only output a single zome. In these kinds of repositories, running `hc-scaffold entry-type` results in `Error: No app manifest (happ.yaml) was found in this directory tree`, because the scaffolding tool first looks for the happ and then the DNA to discover all the zomes in which the entry-type could be created.

This means that I need to keep a `dna.yaml` and a `happ.yaml` files in my repositories which are kind of "fake", as in they are only used to allow the scaffolding tool to discover the only zome in my repository. Maybe if the scaffolding tool could first scan for happs, fall back to find DNAs, and lastly fall back to finding zomes, it would improve the developer experience in these kinds [of repositores](https://github.com/holochain-open-dev/profiles).

What do you think @c12i ?

**Comment by c12i** [comment]
Hi @guillemcordoba

I would like some clarification just to better understand your proposal. At the moment, the repositories scaffolded to be used as modules rather than webhapps still contain a `happ.yaml` and `dna.yaml` despite not really needing one, since the consuming happ would import them as rust dependencies from my interactions with the profiles and file storage zomes.

> Maybe if the scaffolding tool could first scan for happs, fall back to find DNAs, and lastly fall back to finding zome

I am not sure about this, since we would still need to keep the "fake" `happ.yaml` and `dna.yaml` files, which I wonder would really be needed in such a case.

What do you think about having a separate command i.e `hc-scaffold module`?

**Comment by c12i** [comment]
> What would hc scaffold module do?

I'm seeking a better understanding on whether it's still appropriate to use `hc scaffold web-app` to scaffold a module instead of a web application

**Comment by c12i** [comment]
> But why would we need to keep the dna.yaml and happ.yaml files?

I agree with the decision to no longer require the `dna.yaml` and `happ.yaml` files for these kind of repositories. I'm curious about the process for generating the DNA and hApp bundles. I am assuming the bundles would be produced by the web-hApp consuming this module, right?

**Comment by c12i** [comment]
> So in the repository where the zome lives, there is no happ.yaml or dna.yaml but you can still test your app.

Also another question regarding this; the fake manifests getting generated are only useful for testing within the zome's repository context, right?

**Comment by guillemcordoba** [comment]
Well, in the proposed method, you don't need actual dna.yaml or happ.yaml files, you can "fake them" in nix files so that you decide what to include and when. In zome only sweettest, I can create a DNA bundle only with the zome I want to test, which means I can test only that given zome with its integrity. 

> Also another question regarding this; the fake manifests getting generated are only useful for testing within the zome's repository context, right?

Yes exactly, that's the main purpose.

**Comment by c12i** [comment]
I see, so for such repos, the scaffolding tool should not be generating these manifest files, this is the only change required from the cli as far as I understand.

Should "faking" the manifests in nix also be handled by scaffolding as well?

**Comment by guillemcordoba** [comment]
Close, but not really what I mean.

If the command `hc scaffold dna` is run, it's totally fine that a `dna.yaml` file gets created. It is in fact necessary. The thing is that command will never be run in the kinds of repositories we are talking about.

What I'd like to change is that in repos where there's no `dna.yaml`, only zomes, I can't run `hc scaffold entry-type` or `hc scaffold link-type`. This is because the scaffolding tool does not find any happs or DNAs, so it just errors out. I'd like for it to be able to find the zomes that _do_ exist in the repository and choose them as the options where to scaffold the entry or link-type.

I don't think the faking the manifests in nix should be handled by the scaffolding tool no. As a matter of fact I'm scaffolding them myself in the custom template for holochain-open-dev.

**Comment by guillemcordoba** [comment]
Yes same, they will never get crated.

**Comment by guillemcordoba** [comment]
Hmmm how is that going to change the behavior of `hc scaffold entry-type`? The problematic code is [here](https://github.com/holochain/scaffolding/blob/e9f3db78b7ee1c858b29a2b1f65ff05e90b97661/src/cli.rs#L560).

**Comment by c12i** [comment]
> The problematic code is [here](https://github.com/holochain/scaffolding/blob/e9f3db78b7ee1c858b29a2b1f65ff05e90b97661/src/cli.rs#L560).

Yes, this change will also be made,

I am just considering the user who scaffolds a web-app that he intends to use as a module, the project that gets generated should not have these manifest files by default and there won't be any instruction to run `hc scaffold dna`

Do you think this is not necessary and they can manually delete these files to turn their hApp into a module?

**Comment by guillemcordoba** [comment]
Ah no that is already covered with the [module template](https://github.com/holochain-open-dev/templates/tree/main/templates/module), which already scaffolds only the zomes and tells you not to scaffold DNAs. It's cool that this can be done without needing to change the code for the scaffolding tool!

**Comment by c12i** [comment]
Ahh I see, no need to handle this on scaffolding then
